pkg/controller/mpi_job_controller.go

```diff
@@ -346,6 +346,34 @@ func NewMPIJobControllerWithClock(
 
 	controller.updateStatusHandler = controller.doUpdateJobStatus
 
+	// Set up error handlers for informers
+	klog.Info("Setting up informer error handlers")
+	informers := map[string]cache.SharedInformer{
+		"configMapInformer":     configMapInformer.Informer(),
+		"secretInformer":        secretInformer.Informer(),
+		"serviceInformer":       serviceInformer.Informer(),
+		"jobInformer":           jobInformer.Informer(),
+		"podInformer":           podInformer.Informer(),
+		"priorityClassInformer": priorityClassInformer.Informer(),
+		"mpiJobInformer":        mpiJobInformer.Informer(),
+	}
+
+	for name, informer := range informers {
+		err := informer.SetWatchErrorHandler(func(r *cache.Reflector, err error) {
+			// Pipe to default handler first, which just logs the error
+			cache.DefaultWatchErrorHandler(r, err)
+
+			if errors.IsUnauthorized(err) || errors.IsForbidden(err) {
+				klog.Fatalf("Unable to sync cache for informer %s: %s. Requesting controller to exit.", name, err)
```

Comment:
return error here too.

Reply:
oh true.
Still, please don't use `Fatalf`. Follow this guide https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#replacing-fatal-calls

Reply:
To follow this guide, `klog` needs to be upgraded to `klog/v2`. The methods `klog.ErrorS()` and `klog.FlushAndExit()` are not available in `klog/v1.0.0`.

Should `klog` be updated to `klog/v2` for this change?

Reply:
:(

Let's leave that for a follow up

/lgtm
/approve